src/pykeen/stoppers/early_stopping.py

```diff
@@ -157,22 +143,30 @@ def should_stop(self, epoch: int) -> bool:
         )
         result = metric_results.get_metric(self.metric)
 
-        # Only check if enough values are already collected
-        if self.number_evaluations >= self.patience:
-            # Stop if the result did not improve more than delta for patience epochs.
-            if not self.improvement_criterion(buffer=self.buffer, result=result, delta=self.delta):
-                logger.info(f'Stopping early after {self.number_evaluations} evaluations with {self.metric}={result}')
-                for stopped_callback in self.stopped_callbacks:
-                    stopped_callback(self, result, epoch)
-                self.stopped = True
-                return True
-
-        # Update ring buffer
-        self.buffer[self.number_evaluations % self.patience] = result
-        self.number_evaluations += 1
-
         # Append to history
         self.results.append(result)
+        self.number_evaluations += 1
```

Comment:
can't number_evaluations just be calculated as `len(self.results)`? Maybe also set as a property? Or do you prefer to keep it explicit so later it gets dumped with the dataclass?

Reply:
There is already an `number_results` exactly doing this :see_no_evil: I am going to replace `number_evaluations` by that.

Reply:
e443d46